src/form-elements/FormElementRepeatableSet.tsx

```diff
@@ -70,13 +70,13 @@ function FormElementRepeatableSet({
   )
 
   const handleAddEntry = React.useCallback(
-    (index: number | undefined) => {
+    (index: number) => {
       onChange(element, {
         value: (existingEntries) => {
           const newEntries = [...(existingEntries || [])]
           const entry = generateDefaultData(element.elements, {})
 
-          newEntries.splice((index === undefined ? -1 : index) + 1, 0, entry)
+          newEntries.splice(index, 0, entry)
```

Comment:
If you hit the top add button on a multiple add buttons, is the entry inserted at the right place?

Reply:
yep, works like a charm!